lib/delocalize/parameter_delocalizing.rb

```diff
@@ -22,7 +22,16 @@ def delocalize_parse(options, key_stack, value)
     end
 
     def delocalize_parser_for(options, key_stack)
-      parser_type = key_stack.reduce(options) { |h, key| h.stringify_keys[key.to_s] }
+      parser_type = key_stack.reduce(options) do |h, key|
+        break unless h.is_a? Hash
+        h = h.stringify_keys
+        if key =~ /\A-?\d+\z/ && !h.key?(key.to_s)
```

Comment:
Why the need for the `-?` in the regexp?

Also, we should probably extract the `key.to_s` since it's called multiple times. Maybe `key = key.to_s` right after the line with `break`?

Finally, please adhere to the coding style: `h.is_a?(Hash)` and a blank line after the `break`. Thanks. :)